### PR TITLE
Avoid setting routes that route back to the node without its being direct GW to dest

### PR DESCRIPTION
**Pull request by milosgajdos**
Should address https://github.com/micro/development/projects/3#card-25690276

### CODE REVIEW

File: network/default.go [hunk 1]
```diff
@@ -439,11 +439,42 @@ func (n *network) processCtrlChan(l tunnel.Listener) {
...
+					if advertNode.address == "" {
+						advertNode.address = event.Route.Gateway
+					}
+					// if advertising node id is not the same as Route.Router
+					// we know the advertising node is not the origin of the route
+					if advertNode.id != event.Route.Router {
+						// if the origin router is not in the advertising node neighbourhood
+						// we can't rule out potential routing loops so we bail here
+						if _, ok := advertNode.neighbours[event.Route.Router]; !ok {
+							continue
```
Comment (asim) [inline network/default.go:475]: A secondary check here might be necessary. 

If event.Route.Router is in your neighbour list, check if the advert node is in that list. If it's not then you can also ignore.

Comment (milosgajdos) [reply network/default.go:475]: Yeah, thats a good point (checking bidirectionality). I actually had that in my original code before I changed that back to this version and I can't remember now why >_<

Comment (milosgajdos) [reply network/default.go:475]: Oh I know why. I figured if the `event.Route.Router` is in `advertNode` neighbour list that means we can route to `Route.Router` directly from `advertNode`, hence no need to check the other direction. Felt redundant -- as sooner or later they'll announce themselves to each other.

Comment (asim) [reply network/default.go:475]: Wait, here's a question. Does that mean we limit routes that we know of to one hop from our neighbour? Meaning metric of 100. Should we be doing this programmatically by defining a hop count so we actually get to that metric of 1000? Or is this a secondary concern? We know we don't want all the routes but how many do we want?

Comment (milosgajdos) [reply network/default.go:475]: I think having a hop count is an interesting one. So, we can tell hop count up to 2 from the direct neighbourhood list and neighbours' neighbourhood. That is reflected in calculating the Metric in `setRouteMetric`: anything beyond hop count 2 is assigned Metric 1000

Think the idea of allowing a certain size of neighbourhood is an interesting thought to consider.

Comment (asim) [reply network/default.go:475]: I'll make a note rather than adding a card for this and merge the current PR.

**Pull request merged**